### PR TITLE
Parameter expressions

### PR DESCRIPTION
**Pull request by dstepanov**

### CODE REVIEW

**Comment by kareemelzayat** [comment]
I don't know if this is covered here, but I noticed that `literalExpressions` DO NOT get parameterized on Hibernate/JPA; rather they are put literally in the SQL query.

I don't know if that's the whole intent behind the `literal` expression but to me it seems very odd. Why would we want literal data transmitted through the network and risk SQL injections?!

**Comment by dstepanov** [comment]
Are you talking about our criteria implementation or the Hibernate one? Which one does what?

**Comment by kareemelzayat** [comment]
I am really not sure which implementation is causing the issue. But I enabled the SQL logs via `hibernate.show_sql=true` and I could see the literal expression is written as is, instead of a `?`. For example, `CONCAT(entity_.name, testing)` instead of `CONCAT(entity_.name, ?)`

**Comment by dstepanov** [comment]
If the criteria builder is from the Hibernate it's a Hibernate issue

**Comment by kareemelzayat** [comment]
@dstepanov I also suspect so! Do you have a link for the Hibernate repository so I can file an issue there?

**Comment by dstepanov** [comment]
@kareemelzayat https://hibernate.atlassian.net/browse/HHH

File: data-model/src/main/java/io/micronaut/data/model/runtime/QueryParameterBinding.java [hunk 1]
```diff
@@ -147,4 +147,12 @@ default boolean isExpandable() {
     default Object getValue() {
         return null;
     }
+
+    /**
+     * @return Is expression value
+     * @see 4.5.0
```
Comment (radovanradic) [inline data-model/src/main/java/io/micronaut/data/model/runtime/QueryParameterBinding.java:153]:
```suggestion
     * @since 4.7.0
```

**Review by radovanradic — APPROVED** [review]

**Pull request merged**